addons/hw_drivers/controllers/driver.py

```diff
@@ -1,432 +1,382 @@
-# Bluetooth
+# Controllers
 #----------------------------------------------------------
-class GattBtManager(gatt.DeviceManager):
 
-    def device_discovered(self, device):
-        # TODO: need some kind of updated_devices mechanism or not?
-        path = "bt_%s" % (device.mac_address,)
-        if path not in drivers:
-            for driverclass in btdrivers:
-                d = driverclass(device = device, manager=self)
-                if d.supported():
-                    drivers[path] = d
-                    d.connect()
-                    send_iot_box_device(False)
-
-
-class BtManager(Thread):
-    gatt_manager = False
-
-    def run(self):
-        dm = GattBtManager(adapter_name='hci0')
-        self.gatt_manager = dm
-        dm.start_discovery()
-        dm.run()
+class StatusController(http.Controller):
+    @http.route('/hw_drivers/action', type='json', auth='none', cors='*', csrf=False)
+    def action(self, session_id, device_id, data):
+        iot_device = iot_devices.get(device_id)
+        if iot_device:
+            iot_device.action(session_id, data)
+            return True
+        return False
+
+    @http.route('/hw_drivers/event', type='json', auth='none', cors='*', csrf=False)
+    def event(self, listener):
+        """
+        listener is a dict in witch there are a sessions_id and a dict of device_id to listen
+        """
+        req = event_manager.add_request(listener)
+        if req['event'].wait(50):
+            req['event'].clear()
+            req['queue']['session_id'] = req['session_id']
+            return req['queue']
+
+    @http.route('/hw_drivers/box/connect', type='http', auth='none', cors='*', csrf=False)
+    def connect_box(self, token):
+        """
+        This route is called when we want that a IoT Box will be connected to a Odoo DB
+        token is a base 64 encoded string and have 2 argument separate by |
+        1 - url of odoo DB
+        2 - token. This token will be compared to the token of Odoo. He have 1 hour lifetime
+        """
+        server = get_odoo_server_url()
+        image = os.path.join(home(), 'odoo/addons/hw_drivers/static/img/False.jpg')
+        if server == '':
+            token = b64decode(token).decode('utf-8')
+            url, token = token.split('|')
+            if home:
+                path = os.path.join(home(), 'odoo/addons/point_of_sale/tools/posbox/configuration/connect_to_server.sh')
+                try:
+                    subprocess.check_output([path, url, '', token, 'noreboot'])
+                    m.send_alldevices()
+                    image = os.path.join(home(), 'odoo/addons/hw_drivers/static/img/True.jpg')
+                except subprocess.CalledProcessError as e:
+                    _logger.warning(e.output)
+            else:
+                _logger.warning('Home environment variable doesn\'t exist')
+        if os.path.isfile(image):
+            with open(path, 'rb') as f:
+                return f.read()
```

Comment:
if you return here, are you sure the file is correctly closed?
answer: yes :) https://stackoverflow.com/questions/9885217/in-python-if-i-return-inside-a-with-block-will-the-file-still-close